### PR TITLE
gdal 2.6.2: rebuild against libxml2 2.13

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Destination channel:** main

### Links

- [PKG-5283](https://anaconda.atlassian.net/browse/PKG-5283) 

[PKG-5283]: https://anaconda.atlassian.net/browse/PKG-5283?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by skupr-anaconda** [comment]
An error on `win64`:
FileNotFoundError: [WinError 2] The system cannot find the file specified: 'C:\\Windows\\System32\\bash.exe'

It's a bug in `conda-build 24.7.0`, see https://github.com/conda/conda-build/issues/5433

**Comment by SC426** [comment]
Wondering if we can rename the patches to something that's easy for everyone to understand.

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -4,7 +4,3 @@ MACOSX_DEPLOYMENT_TARGET:  # [osx and x86_64]
   - "10.14"                # [osx and x86_64]
 CONDA_BUILD_SYSROOT:       # [osx and x86_64]
   - /opt/MacOSX10.10.sdk   # [osx and x86_64]
-c_compiler:    # [win]
```
Comment (SC426) [inline recipe/conda_build_config.yaml:7]: Was this removed because it's outdated, or for another reason?

Comment (danpetry) [reply recipe/conda_build_config.yaml:7]: yeah, it's in the base cbc.yaml now.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -58,7 +63,7 @@ requirements:
     - openjpeg
     - openssl {{ openssl }}
     - pcre2
-    - poppler
+    - poppler 22.12.0
     - proj
     # qhull disabled because of https://github.com/conda-forge/qgis-feedstock/issues/284#issuecomment-1356490896
```
Comment (danpetry) [inline recipe/meta.yaml:68]: would you be able to unpin the deps that are covered in the base cbc, and replace them with `{{ ... }}`?

Comment (skupr-anaconda) [reply recipe/meta.yaml:68]: Done!

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/conda_build_config.yaml:7]: Yup, it's in the base `cbc.yaml`

**Review by danpetry — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

File: recipe/conda_build_config.yaml
[the remark this answers is not in the document]
Comment (skupr-anaconda) [reply recipe/conda_build_config.yaml:1]: Done

[the remark this answers is not in the document]
Comment (skupr-anaconda) [reply recipe/conda_build_config.yaml:1]: Hmm, osx-64 fails:
```
ld: file not found: /usr/lib/system/libsystem_coretls.dylib for architecture x86_64

clang-14: error: linker command failed with exit code 1 (use -v to see invocation)

make[2]: *** [CMakeFiles/GDAL.dir/build.make:2516: libgdal.32.3.6.2.dylib] Error 1
make[1]: *** [CMakeFiles/Makefile2:4315: CMakeFiles/GDAL.dir/all] Error 2
```

[the remark this answers is not in the document]
Comment (skupr-anaconda) [reply recipe/conda_build_config.yaml:1]: @cbouss I added a comment about MacOSX SDK 10.14

**Pull request merged**